includes/api/class-wc-rest-data-location-controller.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * REST API Data controller
+ *
+ * Handles requests to the /data/location endpoint.
+ *
+ * @author   WooThemes
+ * @category API
+ * @package  WooCommerce/API
+ * @since    2.6.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * REST API Data controller class.
+ *
+ * @package WooCommerce/API
+ * @extends WC_REST_Controller
+ */
+class WC_REST_Data_Location_Controller extends WC_REST_Data_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v2';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'data/locations';
+
+	/**
+	 * Register routes.
+	 *
+	 * @since 3.1.0
+	 */
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			),
+		) );
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<continent>[\w-]+)', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				'args' => array(
+					'continent' => array(
+						'description' => __( 'Continent name to restrict country list', 'woocommerce' ),
+						'type'        => 'string',
+					),
+				),
+			),
+		) );
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<continent>[\w-]+)/(?P<country>[\w-]+)', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				'args' => array(
+					'continent' => array(
+						'description' => __( 'Continent name to restrict country list', 'woocommerce' ),
+						'type'        => 'string',
+					),
+					'country' => array(
+						'description' => __( 'Country name to restrict states list', 'woocommerce' ),
+						'type'        => 'string',
```

Comment:
Correct me if I'm wrong, but I believe we can define valid values for these can we not?